Jenkinsfile

```
@@ -1,72 +1,123 @@
 #!groovy
-
-
-def genBranch(String arch) {
-	return [
-		"${arch}": { ->
-			stage("Build engine image on ${arch}") {
-				wrappedNode(label: "linux&&${arch}", cleanWorkspace: true) {
-					try {
-						checkout scm
-						sh("git clone https://github.com/docker/engine.git engine")
-						sh('make ENGINE_DIR=$(pwd)/engine image')
-					} finally {
-						sh('make ENGINE_DIR=$(pwd)/engine clean-image clean-engine')
+pipeline {
+	agent none
+	options {
+		buildDiscarder(logRotator(daysToKeepStr: '30'))
+		timeout(time: 30, unit: 'MINUTES')
+		timestamps()
+	}
+	parameters {
+		string(name: 'github_repo_engine', defaultValue: 'docker/engine', description: 'github org/repo of engine')
+		string(name: 'github_branch_engine', defaultValue: env.CHANGE_TARGET ?: env.BRANCH_NAME, description: 'github branch of engine')
+		string(name: 'github_repo_cli', defaultValue: 'docker/cli', description: 'github org/repo of cli')
+		string(name: 'github_branch_cli', defaultValue: env.CHANGE_TARGET ?: env.BRANCH_NAME, description: 'github branch of cli')
+		booleanParam(name: 'archive_packages', defaultValue: false, description: 'archive packages')
+	}
+	environment {
+		DOCKER_BUILDKIT = '1'
+	}
+	stages {
+		stage('Build') {
+			parallel {
+				stage('ubuntu-xenial') {
+					agent { label 'amd64 && ubuntu-1804 && overlay2' }
+					stages {
+						stage('create package') {
+							steps {
+								sh 'docker version'
+								sh 'docker info'
+								sh "git clone -b '$params.github_branch_engine' --depth 1 'https://github.com/$params.github_repo_engine' engine"
+								sh "git clone -b '$params.github_branch_cli' --depth 1 'https://github.com/$params.github_repo_cli' cli"
+								sh 'make -C deb VERSION=0.0.1-dev ENGINE_DIR=$(pwd)/engine CLI_DIR=$(pwd)/cli ubuntu-xenial'
+							}
+						}
+						stage('archive package') {
+							when { expression { params.archive_packages } }
+							steps {
+								archiveArtifacts artifacts: 'deb/debbuild/ubuntu-xenial/docker-ce*.deb'
```

Could this use a wildcard for the distro?

```suggestion
								archiveArtifacts artifacts: 'deb/debbuild/**/docker-ce*.deb'
```